proof-systems/src/darlin/poly-commit/src/domain_extended/mod.rs

```diff
@@ -190,4 +232,50 @@ impl<G: Group, PC: PolynomialCommitment<G, Commitment = G>> PolynomialCommitment
     fn challenge_to_scalar(chal: Vec<bool>) -> Result<G::ScalarField, Self::Error> {
         PC::challenge_to_scalar(chal)
     }
+
+    #[cfg(feature = "circuit-friendly")]
+    fn single_point_multi_poly_open<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl IntoIterator<Item=&'a LabeledPolynomial<G::ScalarField>>,
+        point: G::ScalarField,
+        fs_rng: &mut Self::RandomOracle,
+        labeled_randomnesses: impl IntoIterator<Item=&'a LabeledRandomness<Self::Randomness>>,
+        rng: Option<&mut dyn RngCore>
+    ) -> Result<Self::Proof, Self::Error> {
+        /*
+        Iterate over labeled polynomials and values, sorting them in ascending order
+        on the number of segments. The label of the polynomial is employed as a sorting criteria
+        for the polynomials with the same number of segments
+        */
+
+        // employ a counter to check that all polynomials/values are placed in `sorted_collections`,
+        // as in case there are duplicates the `collect` on `BTreeMap`
+        // just stops processing elements of the iterator rather than returning an error
+        let mut counter = 0;
+        let sorted_collections = labeled_polynomials.into_iter().zip(labeled_randomnesses.into_iter()).map(|(poly, rand)| {
+            counter += 1;
+            ((compute_num_of_segments::<G, PC>(ck, poly), poly.label()) , (poly, rand))
+        }
+        ).collect::<BTreeMap<(_, _), (_,_)>>();
+
+        assert_eq!(counter, sorted_collections.len());
+
+        let (sorted_polys, sorted_rands): (Vec<_>, Vec<_>) = sorted_collections.iter().rev().map(|(_, (poly, rand))| (poly, rand)).unzip();
+
+        single_point_multi_poly_open::<G, Self, _, _>(ck, sorted_polys, point, fs_rng, sorted_rands, rng)
+    }
+
+    #[cfg(feature = "circuit-friendly")]
```

Comment:
I wonder if also the primitive can benefit from the re-ordering (the complexity of sorting should be much smaller than the complexity of doing additional arithmetic operations caused by unsorted commitment vecs).
If so, we could re-order a priori, avoid using the circuit-friendly flag, thus reducing code-complexity a bit

Reply:
I did some tests and I observed that the order of segmented commitments has a negligible impact on the performance of the multi-point multi-poly open in the primitive. Therefore, to simplify the code of domain extended poly-commit we decide offline to always sort the commitments with the order employed in the gadget for efficiency.